x-pack/platform/packages/shared/kbn-ai-assistant/src/utils/get_timeline_items_from_conversation.tsx

```diff
@@ -100,18 +96,16 @@ export function getTimelineItemsfromConversation({
       }),
       role: MessageRole.User,
     },
-    ...messagesWithoutSystem.map((message, index) => {
+    ...messages.map((message, index) => {
       const id = v4();
 
       let title: React.ReactNode = '';
       let content: string | undefined;
       let element: React.ReactNode | undefined;
 
       const prevFunctionCall =
-        message.message.name &&
-        messagesWithoutSystem[index - 1] &&
-        messagesWithoutSystem[index - 1].message.function_call
-          ? messagesWithoutSystem[index - 1].message.function_call
+        message.message.name && messages[index - 1] && messages[index - 1].message.function_call
+          ? messages[index - 1].message.function_call
```

Comment:
Could this simply be:
```ts
const prevMessage = messages[index-1];
const prevFunctionCall = prevMessage?.message.function_call
```

or if the check on ` message.message.name` is important:
```ts
const prevMessage = messages[index-1];
const prevFunctionCall = message.message.name && prevMessage?.message.function_call
```